task/prefetch-dependencies-oci-ta/0.1/prefetch-dependencies-oci-ta.yaml

```diff
@@ -211,7 +211,7 @@ spec:
         # 2) '{"packages": [{"type": "gomod"}, {"type": "bundler"}]}'
         # 3) '{"type": "gomod"}'
         function input_json_has_rpm {
-          jq '
+          jq -e '
```

Comment:
Not using it was a deliberate decision based on the following comment: https://github.com/konflux-ci/build-definitions/pull/1850#discussion_r1928624779 . That same PR introduced `set -e` to the bash exec options, so returning 1 means you'd fail the script.

Reply:
They way `input_json_has_rpm` function is currently used (in an `if` condition), it does need to return 1 to work properly.

An alternative would be to switch from `if input_json_has_rpm "$INPUT"` to `if [[ $(input_json_has_rpm "$INPUT") == true ]]`

Reply:
> They way the `input_json_has_rpm` function is currently used (in an `if` condition), it does need to return 1 to work properly.
> 
> An alternative would be to switch from `if input_json_has_rpm "$INPUT"` to `if [[ $(input_json_has_rpm "$INPUT") == true ]]`

Hmm, I stand corrected then.